File: template_frontend/template/src/models/dto/ListUserResponse.ts
export class ListUserResponse {
    id: number;
    name: string;
    email: string;
    profilePicture: string | null;
    active: boolean;
    banned: boolean;
    roles: string[];
    softDeleted: boolean;
    deletedAt: string | null;

    constructor(id: number, name: string, email: string, profilePicture: string|null, active: boolean, banned: boolean, roles: string[], softDeleted: boolean, deletedAt: string|null) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.profilePicture = profilePicture;
        this.active = active;
        this.banned = banned;
        this.roles = roles;
        this.softDeleted = softDeleted;
        this.deletedAt = deletedAt;
    }
}

export default ListUserResponse;

File: template_backend/src/main/java/com/benny1611/template/dto/ListUserResponse.java
package com.benny1611.template.dto;

import lombok.Data;

import java.time.Instant;
import java.util.List;

@Data
public class ListUserResponse {
    private Long id;
    private String name;
    private String email;
    private String profilePicture;
    private boolean active;
    private boolean isBanned;
    private List<String> roles;
    private boolean softDeleted;
    private Instant deletedAt;
}
